tests/components/nexia/test_diagnostics.py

```diff
@@ -0,0 +1,7530 @@
+"""Test august diagnostics."""
+
+from .util import async_init_integration
+
+from tests.components.diagnostics import get_diagnostics_for_config_entry
+
+
+async def test_diagnostics(hass, hass_client):
+    """Test generating diagnostics for a config entry."""
+    entry = await async_init_integration(hass)
+
+    diag = await get_diagnostics_for_config_entry(hass, hass_client, entry)
+    assert diag == {
+        "automations": [
+            {
```

Comment:
Maybe put the data in a json file in a fixtures directory and load it as a fixture?